### PR TITLE
Support for Terraform Cloud / Terraform Enterprise

### PR DESCRIPTION
**Pull request by ayshiff**
This PR adds documentation for [Support for Terraform Cloud / Terraform Enterprise](https://github.com/cloudskiff/driftctl/pull/458).

### CODE REVIEW

**Comment by sundowndev** [comment]
Hey @ayshiff , I just created a branch for v0.8.0, can you rebase your branch to use that branch for your changes? Your PR should only contain changes about the TFC backend docs. Thanks

File: docs/usage/cmd/scan.mdx [hunk 1]
```diff
@@ -56,6 +56,11 @@ $ driftctl scan --from tfstate://my-states/directory
 #
 # You can use the --headers option if you need to add extra headers to the request (e.g: for authentication purposes)
 $ driftctl scan --from tfstate+https://example.com/terraform.tfstate --headers 'Authorization=Basic ...; X-Custom-Header=value'
+
+# You can read the current state version for a Workspace from Terraform Cloud/Terraform Enterprise
+# Don't forget to provide your Terraform Cloud API token
+#
+$ driftctl scan --from tfstate+tfcloud://$WORKSPACE_ID --tfc-token $TFCLOUD_TOKEN
```
Comment (sundowndev) [inline docs/usage/cmd/scan.mdx:63]: May be you use the `TFC_TOKEN` env var directly here

**Review by sundowndev — APPROVED** [review]

File: docs/usage/cmd/scan.mdx [hunk 2]
```diff
@@ -56,6 +56,11 @@ $ driftctl scan --from tfstate://my-states/directory
 #
 # You can use the --headers option if you need to add extra headers to the request (e.g: for authentication purposes)
 $ driftctl scan --from tfstate+https://example.com/terraform.tfstate --headers 'Authorization=Basic ...; X-Custom-Header=value'
+
+# You can read the current state version for a Workspace from Terraform Cloud/Terraform Enterprise
```
Comment (wbeuil) [inline docs/usage/cmd/scan.mdx:60]: What do you mean by `the current state version for a Workspace` ?

Comment (ayshiff) [reply docs/usage/cmd/scan.mdx:60]: I took this from the [Terraform Cloud API documentation](https://www.terraform.io/docs/cloud/api/state-versions.html#fetch-the-current-state-version-for-a-workspace). 

I thought it was pretty self-explanatory for the user. 
For the moment we can only retrieve the current version of the state and not a previous one. (We should use `GET /state-versions/:state_version_id` and ask the user for a specific version if we wanted to do that).

This is just to tell the user that we will fetch the current version of the state.

Comment (wbeuil) [reply docs/usage/cmd/scan.mdx:60]: Ok then let's rewrite it to:

`You can also read the current state for a given workspace from Terraform Cloud / Enterprise`

Comment (wbeuil) [reply docs/usage/cmd/scan.mdx:60]: Don't forget to update the v0.8.0 version also.

**Review by wbeuil — APPROVED** [review]

**Pull request merged**